backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,29 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function putTransaction($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        if ($card->isUserOwner($user)) {
+            return response()->build(self::RESPONSE_MESSAGE_ERROR_UNAUTHORIZED, 'User does not own the card');
+        }
+
+        $data = json_decode(Request::getContent(), true);
+        $card->user_id = $user->id;
+
+        $card->save();
+
+        $transaction = new Transaction();
+        $transaction->card_id = $card_id;
+        $transaction->user_id = $user->id;
```

Comment:
Lets also keep a record of the transaction price (what price the card was sold for).

You can fetch the price of the card from the cards listing.